packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -22,16 +26,27 @@ export class Wallet {
    */
   public static async create(
     engine: Engine,
-    messageService: MessageServiceInterface,
+    messageServiceFactory: MessageServiceFactory,
     retryOptions: Partial<RetryOptions> = DEFAULTS
   ): Promise<Wallet> {
-    return new Wallet(engine, messageService, {...DEFAULTS, ...retryOptions});
+    return new Wallet(engine, messageServiceFactory, {...DEFAULTS, ...retryOptions});
   }
+
+  private _messageService: MessageServiceInterface;
+
   private constructor(
     private _engine: Engine,
-    private _messageService: MessageServiceInterface,
+    messageServiceFactory: MessageServiceFactory,
```

Comment:
Nit, it would be nice to order the parameters so that all private parameters are grouped and all public parameters are grouped.

Reply:
This might be another good thing to enforce with linting if possible.

Reply:
781d334